### PR TITLE
Fix for centos 6.x

### PR DESCRIPTION
**Pull request by wsw0108**

### CODE REVIEW

**Comment by wsw0108** [comment]
`CAIRO_FORMAT_RGB16_565` was added since cairo-v1.2.0, but mark as DEPRECATED from next version cairo-v1.2.2, until cairo-v1.10.0

**Comment by wsw0108** [comment]
related to #1075

**Comment by zbjornson** [comment]
Cairo has built-in version macros actually: https://cairographics.org/manual/cairo-Version-Information.html

```cpp
##if CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1, 0, 0)
printf ("Compiling with suitable cairo version: %s\n", %CAIRO_VERSION_STRING);
##endif

if (cairo_version() >= CAIRO_VERSION_ENCODE(1, 0, 0))
    printf ("Running with suitable cairo version: %s\n", cairo_version_string ());
```

Want to use that instead?

**Comment by wsw0108** [comment]
I tried use the version macros from `cairo.h`, but it cause error on ci.

File: src/CanvasRenderingContext2d.cc [hunk 1]
```diff
@@ -635,7 +635,9 @@ NAN_METHOD(Context2d::New) {
         if (!strcmp(*utf8PixelFormat, "RGBA32")) format = CAIRO_FORMAT_ARGB32;
         else if (!strcmp(*utf8PixelFormat, "RGB24")) format = CAIRO_FORMAT_RGB24;
         else if (!strcmp(*utf8PixelFormat, "A8")) format = CAIRO_FORMAT_A8;
+#if CAIRO_VERSION >= CAIRO_VERSION_ENCODED(1, 10, 0)
```
Comment (zbjornson) [inline src/CanvasRenderingContext2d.cc]: The macro is `ENCODE`, not `ENCODED` -- hopefully that fixes it.

**Comment by wsw0108** [comment]
I make this PR because some cloud host like aliyun still provide CentOS-6 as guest os.

**Comment by LinusU** [comment]
A downside with just `ifdef`ing these out is that passing `RGB16_565` will create a canvas using `CAIRO_FORMAT_RGB24` without giving any indication to the user that something went wrong.

If we are going down this route, I think that we need to add `throw`s when the library is called with something that we cannot support 🤔

**Comment by wsw0108** [comment]
I agree with you. 
I will add some check for this.

**Comment by wsw0108** [comment]
For now I think there is no reason to support old CentOS6.x, so I will close this.

**Pull request closed without merging**